docs/architecture/integrations/filecoin/filecoin-storage.md

```diff
@@ -0,0 +1,104 @@
+
+Currently Nevermined integrated with the most popular centralized/cloud based storage providers (Amazon S3, Azure, etc.). This document details the integration Nevermined with Filecoin allowing to:
+
+- Use Filecoin as one of the options supported allowing Nevermined users to publish & share their data via Filecoin
+- Facilitate access control & data monetization of Filecoin existing data
+- Enable a decentralized storage solution where users don't need to upload their data to any existing centralized solution
+
+
+![Nevermined & Filecoin High Level integration](images/integration-high-level.png)
+
+## Value
+
+The integration of Filecoin into Nevermined brings is valuable for Filecoin community because:
+
+- Increase the usage of Filecoin network
+- Provides utility to Filecoin network via integration with existing and mature Open Source software
+- Filecoin doesn't provide a granular access control allowing to the data owners or publishers to decided when their data can be accessible
+- Allows user using centralized data storage based on cloud providers to use Filecoin as alternative
+- Nevermined is L2 solution, network independent, and can be deployed in public or private networks. Via this integration, Filecoin could be used in any Nevermined user deployment
+- Nevermined provides compute to the data and provenance (based on W3C PROV) capabilities. Via the integration, the Filecoin community would be able to use high value capabilities on top of their data
+
+## Integration Details
+
+The integration of Filecoin as a fully supported storage provider require the modification and delivery of the following components:
+
+* The [Nevermined Gateway](https://github.com/nevermined-io/gateway). This component is in charge of making available Nevermined users data. Before this integration the gateway supports different storage providers (Amazon S3, Azure, On Premise, etc.). This integration adds support to Filecoin as storage mechanism allowing to decrypt user urls and resolve Filecoin CIDs.
+* The Nevermined SDKs. To facilitate user adoption, Nevermined support SDKs in 3 different programming languages:
+  - [Javascript SDK](https://github.com/nevermined-io/sdk-js), to facilitate the integration of Nevermined in web interfaces
+  - [Python SDK](https://github.com/nevermined-io/sdk-py), to facilitate the integration of Nevermined in data science tools
+  It will deliver a modification of the 3 SDKs allowing to the users to publish in Nevermined Filecoin contents (CIDs)  
```

Comment:
```suggestion
  - [Java SDK](https://github.com/nevermined-io/sdk-java), to facilitate the integration of Nevermined with existing industry big data solutions. 
- It will deliver a modification of the 3 SDKs allowing the users to publish in Nevermined,  Filecoin contents (CIDs)  
```